File: backend/app/schemas/retrieval_log.py
from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class RetrievalLogSummary(BaseModel):
    """List view for a retrieval log row (no full packet payload)."""

    id: str
    question: str
    confidence: float | None = None
    warnings: list[str] | None = None
    latency_ms: int | None = None
    token_estimate: int | None = None
    evidence_unit_count: int = 0
    has_retrieval_packet: bool = False
    created_at: datetime


class RetrievalLogListResponse(BaseModel):
    """Paginated list of retrieval logs (newest first)."""

    items: list[RetrievalLogSummary]


class RetrievalLogDetailResponse(BaseModel):
    """Single retrieval log including optional canonical `retrieval_packet` JSON."""

    id: str
    question: str
    selected_indexes: list[dict[str, Any]] | None = None
    graph_path: dict[str, Any] | None = None
    evidence_unit_ids: list[str] | None = None
    confidence: float | None = None
    warnings: list[str] | None = None
    latency_ms: int | None = None
    token_estimate: int | None = None
    metadata: dict[str, Any] | None = Field(
        default=None,
        description="Row metadata (DB column `metadata`).",
    )
    retrieval_packet: dict[str, Any] | None = None
    created_at: datetime

File: frontend/src/types/retrievalLog.ts
/**
 * Mirrors `backend/app/schemas/retrieval_log.py`.
 * Datetimes are ISO strings in JSON.
 */

export interface RetrievalLogSummary {
  id: string;
  question: string;
  confidence: number | null;
  warnings: string[] | null;
  latency_ms: number | null;
  token_estimate: number | null;
  evidence_unit_count: number;
  has_retrieval_packet: boolean;
  created_at: string;
}

export interface RetrievalLogListResponse {
  items: RetrievalLogSummary[];
}

export interface RetrievalLogDetailResponse {
  id: string;
  question: string;
  selected_indexes: Record<string, unknown>[] | null;
  graph_path: Record<string, unknown> | null;
  evidence_unit_ids: string[] | null;
  confidence: number | null;
  warnings: string[] | null;
  latency_ms: number | null;
  token_estimate: number | null;
  metadata: Record<string, unknown> | null;
  retrieval_packet: Record<string, unknown> | null;
  created_at: string;
}
